extension/app/ts/background/popupMessageHandlers.ts

```diff
@@ -169,6 +169,18 @@ export async function refreshSimulation(ethereumClientService: EthereumClientSer
 	}, settings.activeSimulationAddress)
 }
 
+export async function refreshPopupConfirmTransactionMetadata(simulator: Simulator, userAddressBook: UserAddressBook, { data }: RefreshConfirmTransactionMetadata) {
+	console.log('refreshPopupConfirmTransactionMetadata')
+	const addressMetadata = await getAddressBookEntriesForVisualiser(simulator, data.visualizerResults.map((x) => x.visualizerResults), data.simulationState, userAddressBook)
+	return await sendPopupMessageToOpenWindows({
+		method: 'popup_confirm_transaction_simulation_state_changed' as const,
+		data: {
+			...data,
+			addressBookEntries: addressMetadata,
+		}
```

Comment:
Style Suggestion: Consider one line here, doesn't feel like we are improving readability in this case.  For consistency, consider this style when all you are doing is a copy plus one addition/change as done here.
```suggestion
		data: { ...data, addressBookEntries: addressMetadata }
```